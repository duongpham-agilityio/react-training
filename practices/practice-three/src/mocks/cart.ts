import { TCartItemProps } from '@/components/Cart/components';

export const cartItemProps: TCartItemProps = {
  productId: 1,
  description: 'Adidas Menâ€™s T-Shirt',
  name: 'Adidas T-Shirt',
  imageURL:
    'https://static.nike.com/a/images/t_PDP_1280_v1/f_auto,q_auto:eco/d455e5b6-23e9-4f06-8b86-32610d838937/renew-run-3-road-running-shoes-395S7c.png',
  price: 200,
  quantity: 2,
  onChangeQuantity: () => {},
  onRemove: () => {},
};
