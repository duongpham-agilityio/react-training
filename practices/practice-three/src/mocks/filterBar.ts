// Types
import { FilterBarProps } from '@/components';

export interface IFilterOption {
  name: string;
  value: string;
}

export const filterOptions: IFilterOption[] = [
  {
    name: 'All',
    value: 'all',
  },
  {
    name: 'Fashion',
    value: 'fashion',
  },
  {
    name: 'Electrics',
    value: 'electrics',
  },
  {
    name: 'Books',
    value: 'books',
  },
];

export const filterBarProps: FilterBarProps = {
  value: '',
  onChange: () => {},
};
