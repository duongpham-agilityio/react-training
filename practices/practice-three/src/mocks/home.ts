import { IProduct } from '@/interface';

export const products: IProduct[] = [
  {
    name: 'Tasty Plastic Mouse',
    description: 'Quos tempora adipisci quidem ipsum.',
    imageURL: 'https://loremflickr.com/640/480/fashion',
    price: 430,
    quantity: 10,
    isLiked: true,
    id: 1,
    category: 'fashion',
    createdAt: new Date(),
  },
  {
    name: 'Tasty Wooden Chair',
    description:
      'Ab repellat ullam autem. Esse facere facere debitis dicta quidem nihil. Saepe suscipit ex.',
    imageURL: 'https://loremflickr.com/640/480/fashion',
    price: 430,
    quantity: 10,
    isLiked: false,
    id: 2,
    category: 'fashion',
    createdAt: new Date(),
  },
];
