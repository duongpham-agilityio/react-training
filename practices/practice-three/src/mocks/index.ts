export * from './productCard';
export * from './filterBar';
export * from './inputForm';
export * from './pagination';
