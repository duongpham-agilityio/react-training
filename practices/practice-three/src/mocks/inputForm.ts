// Components
import { TInputFormProps } from '@/components/common';

export const inputFormProps: TInputFormProps = {
  label: 'Email',
  value: '',
  placeholder: 'Please enter email',
  onChange: () => {},
};
