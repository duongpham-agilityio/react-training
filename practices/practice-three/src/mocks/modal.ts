import { TModalProps } from '@/components';

export const defaultModalProps: TModalProps = {
  title: 'Demo',
  isOpen: true,
  onClose: () => {},
};
