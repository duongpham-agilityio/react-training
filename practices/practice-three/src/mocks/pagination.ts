import { PaginationProps } from '@/components';

export const defaultPropsPagination: PaginationProps = {
  data: [1, 2, 3, 4, 5, 6, 7],
  currentPage: 1,
  isNextPage: true,
  isPrevPage: false,
  onChangePage: () => {},
  onNextPage: () => {},
  onPreviousPage: () => {
    console.log(122);
  },
};
