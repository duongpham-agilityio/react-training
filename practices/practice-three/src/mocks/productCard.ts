import { ProductCardProps } from '@/components';

export const productCardProps: ProductCardProps = {
  info: {
    imageURL:
      'https://static.nike.com/a/images/t_PDP_1280_v1/f_auto,q_auto:eco/d455e5b6-23e9-4f06-8b86-32610d838937/renew-run-3-road-running-shoes-395S7c.png',
    title: 'Nike React Miler',
    description: 'Apple M1 Chip with 8‑Core CPU and 8‑Core GPU 512GB Storage',
    id: 1,
    price: 1000,
    status: false,
    isLike: false,
    statusMessage: 'Only 5 left',
  },
  onAddToCard: () => {},
  onLike: () => {},
};
