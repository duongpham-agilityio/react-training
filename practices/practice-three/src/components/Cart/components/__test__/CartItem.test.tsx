import { fireEvent, render } from '@testing-library/react';

// Components
import { CartItem } from '..';

// Mocks
import { cartItemProps } from '@/mocks';

const setup = (props = cartItemProps) => render(<CartItem {...props} />);

describe('CartItem', () => {
  it('Match top snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it('Re-render', () => {
    const { rerender } = setup();

    rerender(
      <CartItem
        {...{
          ...cartItemProps,
          data: {
            ...cartItemProps,
            name: 'Duong',
          },
        }}
      />,
    );
  });

  it('Actions', () => {
    const onChangeQuantity = jest.fn();
    const onRemove = jest.fn();
    const { getAllByRole } = setup({
      ...cartItemProps,
      onChangeQuantity,
      onRemove,
    });
    const buttons = getAllByRole('button');
    const increaseBtn = buttons[1];
    const decreaseBtn = buttons[0];
    const removeBtn = buttons[2];

    fireEvent.click(decreaseBtn);
    expect(onChangeQuantity).toBeCalled();

    fireEvent.click(increaseBtn);
    expect(onChangeQuantity).toBeCalledTimes(2);

    fireEvent.click(removeBtn);
    expect(onRemove).toBeCalled();
  });
});
