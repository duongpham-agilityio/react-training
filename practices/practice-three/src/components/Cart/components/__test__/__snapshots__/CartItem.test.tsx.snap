// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem Match top snapshot 1`] = `
<div>
  <div
    class="css-6havx5"
  >
    <div
      class="css-13hz88"
    >
      <img
        class="chakra-image css-exprrl"
        src="https://static.nike.com/a/images/t_PDP_1280_v1/f_auto,q_auto:eco/d455e5b6-23e9-4f06-8b86-32610d838937/renew-run-3-road-running-shoes-395S7c.png"
      />
      <div
        class="css-0"
      >
        <h2
          class="chakra-heading css-navt0c"
        >
          Adidas T-Shirt
        </h2>
        <p
          class="chakra-text css-1r2n17h"
        >
          Adidas Menâ€™s T-Shirt
        </p>
      </div>
    </div>
    <div
      class="css-17xq282"
    >
      <div
        class="css-1en42kf"
      >
        <button
          aria-label="Button decrease quantity"
          class="chakra-button css-p7mkev"
          type="button"
        >
          <svg
            aria-hidden="true"
            fill="#000"
            focusable="false"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m3.86 8.753 5.482 4.796c.646.566 1.658.106 1.658-.753V3.204a1 1 0 0 0-1.659-.753l-5.48 4.796a1 1 0 0 0 0 1.506z"
            />
          </svg>
        </button>
        <p
          class="chakra-text css-0"
        >
          2
        </p>
        <button
          aria-label="Button increase quantity"
          class="chakra-button css-p7mkev"
          type="button"
        >
          <svg
            aria-hidden="true"
            fill="#000"
            focusable="false"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m12.14 8.753-5.482 4.796c-.646.566-1.658.106-1.658-.753V3.204a1 1 0 0 1 1.659-.753l5.48 4.796a1 1 0 0 1 0 1.506z"
            />
          </svg>
        </button>
      </div>
      <p
        class="chakra-text css-1nwj029"
      >
        $
        200
      </p>
      <div
        class="css-1en42kf"
      >
        <button
          aria-label="Button remove from cart"
          class="chakra-button css-p7mkev"
          type="button"
        >
          <svg
            aria-hidden="true"
            fill="#000"
            focusable="false"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.5 5.5A.5.5 0 0 1 6 6v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5Zm2.5 0a.5.5 0 0 1 .5.5v6a.5.5 0 0 1-1 0V6a.5.5 0 0 1 .5-.5Zm3 .5a.5.5 0 0 0-1 0v6a.5.5 0 0 0 1 0V6Z"
            />
            <path
              d="M14.5 3a1 1 0 0 1-1 1H13v9a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V4h-.5a1 1 0 0 1-1-1V2a1 1 0 0 1 1-1H6a1 1 0 0 1 1-1h2a1 1 0 0 1 1 1h3.5a1 1 0 0 1 1 1v1ZM4.118 4 4 4.059V13a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4.059L11.882 4H4.118ZM2.5 3h11V2h-11v1Z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
